code/datums/components/material_container.dm

```diff
@@ -83,46 +82,59 @@
 	set waitfor = FALSE
 	var/requested_amount
 	var/active_held = user.get_active_held_item()  // differs from I when using TK
-	if(istype(I, /obj/item/stack) && precise_insertion)
-		var/atom/current_parent = parent
-		var/obj/item/stack/S = I
-		requested_amount = input(user, "How much do you want to insert?", "Inserting [S.singular_name]s") as num|null
-		if(isnull(requested_amount) || (requested_amount <= 0))
-			return
-		if(QDELETED(I) || QDELETED(user) || QDELETED(src) || parent != current_parent || user.physical_can_use_topic(current_parent) < UI_INTERACTIVE || user.get_active_held_item() != active_held)
-			return
+	if(istype(I, /obj/item/stack))
+		if(precise_insertion)
+			var/atom/current_parent = parent
+			var/obj/item/stack/S = I
+			requested_amount = input(user, "How much do you want to insert?", "Inserting [S.singular_name]s") as num|null
+			if(isnull(requested_amount) || (requested_amount <= 0))
+				return
+			if(QDELETED(I) || QDELETED(user) || QDELETED(src) || parent != current_parent || user.physical_can_use_topic(current_parent) < UI_INTERACTIVE || user.get_active_held_item() != active_held)
+				return
+		else
+			if((total_amount + get_item_material_amount(I)) <= max_amount)
+				requested_amount = get_item_material_amount(I)
+			else
+				requested_amount = (max_amount-total_amount)
 	if(!user.temporarilyRemoveItemFromInventory(I))
```

Comment:
I understand this line to be the problem but I'm not sure how to fix it without removing the functionality, which I assume is for making sure you don't insert TRAIT_NODROP items.